haystack/document_stores/elasticsearch.py

```diff
@@ -495,11 +114,46 @@ def _create_label_index(self, index_name: str, headers: Optional[Dict[str, str]]
             if not self._index_exists(index_name, headers=headers):
                 raise e
 
+    def _get_raw_similarity_score(self, score):
+        return score - 1000
+
+    def _get_vector_similarity_query(self, query_emb: np.ndarray, top_k: int):
+        """
+        Generate Elasticsearch query for vector similarity.
+        """
+        if self.similarity == "cosine":
+            similarity_fn_name = "cosineSimilarity"
+        elif self.similarity == "dot_product":
+            similarity_fn_name = "dotProduct"
+        elif self.similarity == "l2":
+            similarity_fn_name = "l2norm"
+        else:
+            raise DocumentStoreError(
+                "Invalid value for similarity in ElasticSearchDocumentStore constructor. Choose between 'cosine', 'dot_product' and 'l2'"
+            )
```

Comment:
Can we maybe do this conversion directly in `SearchEngineDocumentStore.__init__`? It's already checking that it's one of those values so might as well convert it right?